src/components/pages/Billing.tsx

```
@@ -0,0 +1,209 @@
+import React, { useEffect, useMemo, useState } from "react";
+import { Card } from "@/components/ui/card";
+import { useToast } from "@/hooks/use-toast";
+import { supabase } from "@/integrations/supabase/client";
+import { useModels } from "@/hooks/useModels";
+import { useTimeRange } from "@/hooks/useTimeRange";
+import {
+  ChartContainer,
+  ChartTooltip,
+  ChartTooltipContent,
+} from "@/components/ui/chart";
+import {
+  ResponsiveContainer,
+  AreaChart,
+  Area,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+} from "recharts";
+
+type PlatformRow = {
+  id: string;
+  name: string;
+  model: string;
+  input_cost: number | null;   // $ per 1K input tokens
+  output_cost: number | null;  // $ per 1K output tokens
+};
+
+type RunRow = {
+  run_at: string;
+  input_tokens: number | null;
+  output_tokens: number | null;
+};
+
+type DailyCost = {
+  day: string;     // YYYY-MM-DD
+  cost: number;    // USD
+  inTok: number;   // summed tokens (optional for debugging/extension)
+  outTok: number;
+};
+
+const toISO = (d: Date) => new Date(d).toISOString();
```

Using toISOString() here coerces local times to UTC, which can shift your gte/lte range and silently drop runs near midnight (e.g., “yesterday”/“today”). Consider normalizing to explicit UTC day boundaries and using a half-open range.